chapters/missing-data.qmd

```diff
@@ -0,0 +1,598 @@
+
+When the predictor that requires imputation is categorical, then logistic regression can be used to generate an imputed category.
+
+### Nearest Neighbors
+
+$K$-nearest neighbor calculations were previously discussed in @sec-mds where it was used in some multidimensional scaling methods. When imputing a missing predictor value for a new sample, the $K$ most similar samples from the training set are determined (using all complete predictor values). The $K$ predictor values for the predictor of interest are summarized via the mean (for numeric outcomes) or the mode. This summary statistic is used to fill in the missing value. 
+
+### Trees
+
+As mentioned earlier, tree-based models are a reasonable choice for imputation techniques because many types of trees do not require complete data themselves. They generally provide good accuracy and do not extrapolate values beyond the bounds of the training data. While using a single tree can be used for imputation, we know that the resulting imputation will likely have low bias and high variance. Ideally, we would like imputed values would have low bias and low variance.
+
+Tree ensembles, such as bagging and random forests (Sections X and Y), help solve this issue since they blend the predictors from many individual tree models. However, these methods can be computationally taxing for moderate to large size data sets. Specifically, random forests require a large number of trees (hundreds to thousands) to achieve a stable and reliable imputation model. This comes at a cost of a large computational footprint which may become a challenge as the number of predictors with missing data increases; a separate model must be trained and retained for each predictor. As we will discuss later, bagged tree models tend to be smaller and faster than their random forest counterparts. 
+
+Like nearest neighbor imputation, the bagged tree also places most of the imputed values close to the distribution of the original data as see in @fig-imputation-comparison. The distribution of distances (@fig-imputation-distance-comparison) for bagged imputation is similar to nearest neighbors.
+
+Which imputation technique we choose depends on the problem. If a small number of predictors have missing values, then using $K$-NN or bagged imputation may not add much computational burden to the modeling process. However, if many predictors have missing values and the data set is large, then the model based imputation techniques may become cumbersome. In this case, it may be prudent to begin with a most likely value imputation approach. If the predictive performance of the optimal tuned machine learning technique is high, then the imputation approach was sufficient. However, if predictive performance is lagging, then we could implement a different imputation technique while training models. 
+
+TODO IDK
```

Comment:
> it may be prudent to begin with a most likely value imputation approach

I don't know that I agree with this. I'll wordsmith it a little and you can let me know what you think (or if I'm overthinking it).

Reply:
That wording might not be the best here.  What I'm trying to convey is that if there will be a high computational burden already (without layering on the additional computational burden of imputation), and the amount of missing data is small, then using a simple imputation technique may be sufficient and will likely not harm predictive performance.  This has been my experience so far.

Reply:
cad2d00